src/parse.d

```diff
@@ -1100,7 +1100,6 @@ public:
             TemplateParameters* tpl = null;
             if (token.value == TOKlparen)
                 tpl = parseTemplateParameterList();
-            assert(token.value == TOKassign);
             nextToken(); // skip over '='
```

Comment:
> Doesn't that mean that malformed user input may cause the following code to malfunction?

Parser should accept any possible token streams. If the input tokens could be treated as valid D code representation, parser will generate AST. If not, parser will report some parsing errors. Both are valid as "parser program output".
In other words, any input token streams must not be treated as malformed. If any assumptions exist for the input token streams, it's just wrong. In here, I've just removed wrong assertion.

> Shouldn't this be turned into a check with a compile error message when the next token isn't = instead?

In `parseTemplateParameterList` function, it's already reported errors for the parsing failures. But, indeed it would be better to have own error check here. I'll replace the bare `nextToken()` call with `check(TOKassign)`.